explorer/client/Explorer.tsx

```diff
@@ -45,91 +48,60 @@ import { EntityPicker } from "grapher/controls/entityPicker/EntityPicker"
 
-export interface ExplorerProps {
-    explorerProgramCode: string
-    slug: string
-    explorerProgram?: ExplorerProgram
+export interface ExplorerProps extends SerializedExplorerProgram {
+    explorerProgram?: ExplorerProgram // todo: why do we need this? IIRC it had something to do with speeding up the create page
     chartConfigs?: GrapherInterface[]
     bindToWindow?: boolean
     queryString?: string
     isEmbed?: boolean // todo: what specifically does this mean? Does it mean IFF in an iframe? Or does it mean in an iframe OR hoisted?
     globalEntitySelection?: GlobalEntitySelection // todo: use this
 }
 
-interface BootstrapProps {
-    slug: string
-    explorerProgramCode: string
-    containerNode: HTMLElement
-    isEmbed?: boolean // todo: what specifically does this mean? Does it mean IFF in an iframe? Or does it mean in an iframe OR hoisted?
-    queryStr?: string
-    globalEntitySelection?: GlobalEntitySelection
-    bindToWindow?: boolean
-}
-
 @observer
 export class Explorer
     extends React.Component<ExplorerProps>
-    implements ObservableUrl, SlideShowManager, EntityPickerManager {
+    implements SlideShowManager, EntityPickerManager {
     static bootstrap(props: ExplorerProps) {
         return ReactDOM.render(
             <Explorer {...props} queryString={window.location.search} />,
             document.getElementById(ExplorerContainerId)
         )
     }
 
-    static async createExplorerAndRenderToDom(props: BootstrapProps) {
-        return ReactDOM.render(
-            <Explorer
-                explorerProgramCode={props.explorerProgramCode}
-                slug={props.slug}
-                queryString={props.queryStr}
-                isEmbed={props.isEmbed}
-                globalEntitySelection={props.globalEntitySelection}
-                bindToWindow={props.bindToWindow}
-            />,
-            props.containerNode
-        )
+    static async createExplorerAndRenderToDom(
+        props: ExplorerProps,
+        containerNode: HTMLElement
+    ) {
+        return ReactDOM.render(<Explorer {...props} />, containerNode)
     }
 
     private urlBinding?: UrlBinder
 
     @computed private get explorerProgram() {
-        return (
-            this.props.explorerProgram ??
-            new ExplorerProgram(
-                this.props.slug,
-                this.props.explorerProgramCode,
-                this.props.queryString
-            )
+        const program =
+            this.props.explorerProgram ?? ExplorerProgram.fromJson(this.props)
+        program.decisionMatrix.setValuesFromQueryString(
+            this.props.queryString ?? program.defaultView
         )
+        return program
     }
 
-    @observable hideControls = false
     @observable entityPickerMetric?: ColumnSlug =
-        strToQueryParams(this.explorerProgram.queryString ?? "").pickerMetric ??
-        undefined
+        strToQueryParams(this.props.queryString ?? "").pickerMetric ?? undefined
```

Comment:
Ah, just remembered I need to put pickerMetric into query string (And I'll add typing for this). There is a notion issue already